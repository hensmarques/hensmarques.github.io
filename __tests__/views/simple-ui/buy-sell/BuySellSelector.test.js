import React from 'react';
import BuySellSelector from '../../../../src/views/simple-ui/buy-sell/BuySellSelector';

describe('<BuySellSelector/>', () => {
  const currencyName = 'BTC';
  const onChange = jest.fn();

  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<BuySellSelector currencyName={currencyName} onChange={onChange} value="buy" />);
    expect(wrapper).toMatchSnapshot();    
  });

  it('renders with the select-indicator on the sell side', () => {
    const value = 'sell';
    const wrapper = shallow(<BuySellSelector currencyName={currencyName} onChange={onChange} value={value} />);
    expect(wrapper.find(`.${value}-selector`).hasClass('active')).toBe(true);
    const sellSelector = wrapper.find('.sell-selector');
    const buySelector = wrapper.find('.buy-selector');
    expect(sellSelector.find('.select-indicator').exists()).toBe(true);
    expect(buySelector.find('.select-indicator').exists()).toBe(false);
  });
});
