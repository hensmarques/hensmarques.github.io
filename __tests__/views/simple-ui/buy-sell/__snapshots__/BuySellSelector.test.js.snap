// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuySellSelector/> renders and matches the snapshot 1`] = `
<div
  className="buy-sell-selector"
>
  <div
    className="buy-selector option active"
  >
    Buy BTC
    <div
      className="select-indicator"
    />
  </div>
  <div
    className="sell-selector option "
  >
    Sell BTC
  </div>
</div>
`;
